examples/virtual-device-app/android/App/feature/closure/src/main/java/com/matter/virtual/device/app/feature/closure/DoorLockViewModel.kt

```diff
@@ -1,17 +1,104 @@
 package com.matter.virtual.device.app.feature.closure
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.viewModelScope
+import com.matter.virtual.device.app.core.common.successOr
+import com.matter.virtual.device.app.core.domain.usecase.matter.IsFabricRemovedUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.StartMatterAppServiceUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.cluster.doorlock.GetLockStateFlowUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.cluster.doorlock.SendLockAlarmEventUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.cluster.doorlock.SetLockStateUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.cluster.powersource.GetBatPercentRemainingUseCase
+import com.matter.virtual.device.app.core.domain.usecase.matter.cluster.powersource.SetBatPercentRemainingUseCase
 import com.matter.virtual.device.app.core.ui.BaseViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
 import timber.log.Timber
 
 @HiltViewModel
-class DoorLockViewModel @Inject constructor(savedStateHandle: SavedStateHandle) :
-  BaseViewModel(savedStateHandle) {
+class DoorLockViewModel
+@Inject
+constructor(
+  getLockStateFlowUseCase: GetLockStateFlowUseCase,
+  getBatPercentRemainingUseCase: GetBatPercentRemainingUseCase,
+  private val setLockStateUseCase: SetLockStateUseCase,
+  private val sendLockAlarmEventUseCase: SendLockAlarmEventUseCase,
+  private val setBatPercentRemainingUseCase: SetBatPercentRemainingUseCase,
+  private val startMatterAppServiceUseCase: StartMatterAppServiceUseCase,
+  private val isFabricRemovedUseCase: IsFabricRemovedUseCase,
+  savedStateHandle: SavedStateHandle
+) : BaseViewModel(savedStateHandle) {
+
+  private val _lockState: StateFlow<Boolean> = getLockStateFlowUseCase()
+  val lockState: LiveData<Boolean>
+    get() = _lockState.asLiveData()
+
+  private val _batteryStatus: MutableStateFlow<Int> =
```

Comment:
This name seems not ideal. How about `_batteryPercentage` and similar for the getters as this seems to be the content? if this is a status, a typesafe value seems better.

Overall I think we should have `isLocked` and `batteryFillPercentage` instead of the existing `lockState` and `batteryStatus`.

Reasoning is that answering this is hard:
   - For a random class, lockState is `true`. Is it locked or unlocked?
   - For a random class, batteryStatys is `49`. What is its percentage, is it charging or discharging (since it is a state after all) ?
   
 Where as these are obvious:
    - For a random class, `isLocked` is true. Is it locked or unlocked?
    - For a random class, `batteryFillPercentage` is 39. What is its fill? Do you know if it is charging or discharging?

Reply:
> This name seems not ideal. How about `_batteryPercentage` and similar for the getters as this seems to be the content? if this is a status, a typesafe value seems better.
> 
> Overall I think we should have `isLocked` and `batteryFillPercentage` instead of the existing `lockState` and `batteryStatus`.
> 
> Reasoning is that answering this is hard:
> 
> * For a random class, lockState is `true`. Is it locked or unlocked?
> * For a random class, batteryStatys is `49`. What is its percentage, is it charging or discharging (since it is a state after all) ?
> 
> Where as these are obvious:
> 
> * For a random class, `isLocked` is true. Is it locked or unlocked?
> * For a random class, `batteryFillPercentage` is 39. What is its fill? Do you know if it is charging or discharging?

Thank you for your detailed review. I update some patches to use more obvious name.
* Use `LockState` enum
* Use `batterRemainingPercentage`
(to align with the specification as much as possible)